File: eshop-parent/eshop-goods-interface/src/main/java/com/eshop/goods/service/SpecificationService.java
package com.eshop.goods.service;

import java.util.List;

import com.eshop.common.PageResult;
import com.eshop.pojo.TbSpecification;
import com.eshop.specification.pojo.Specification;

/**
 * 规格接口
 * @author yzm
 *
 */
public interface SpecificationService {
	
	public List<TbSpecification> findAll();   
	
	/**
	 * 品牌分页
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	/**
	 * 根据Id查询实体
	 * @param id
	 * @return
	 */
	public TbSpecification findOne(Long id);
	
	/**
	 * 根据查询的实体进行修改
	 * @param specification
	 * @return
	 */
	public void update(TbSpecification specification);
	
	/**
	 * 删除
	 * @param id
	 */
	public void delete(Long[] ids);
	
	/**
	 * 根据查询条件查询
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbSpecification specification,int pageNum,int pageSize);
	
	/**
	 * 新增规格
	 * @param specification 规格pojo
	 */
	public void add(Specification specification);
}

File: eshop-parent/eshop-manager-web/src/main/webapp/js/service/specificationService.js
app.service('specificationService',function($http){
    	//读取列表数据绑定到表单中
    	this.findAll=function(){
    		return $http.get('../specification/findAll.do');		
    	}
    	//分页 
    	this.findPage=function(page,size){
    		return $http.get('../specification/findPage.do?pageNum='+page+'&pageSize='+size);		
    	}
    	
    	//查询实体
    	this.findOne=function(id){
    		return $http.get('../specification/findOne.do?id=' + id);		
    	}    
    	
    	//增加 
    	this.add=function(entity){
    		return $http.post('../specification/add.do',entity);		
    	}
    	
    	//更新
    	this.update=function(entity){
    		return $http.post('../specification/update.do',entity);		
    	}
    	
    	//删除
    	this.del=function(ids){
    		return $http.get('../specification/delete.do?ids=' + ids);		
    	}

    	//搜索
    	this.search=function(page,size,searchEntity){
    		return $http.post('../specification/search.do?page='+page +'&size='+size,searchEntity);		
    	}
	}); 